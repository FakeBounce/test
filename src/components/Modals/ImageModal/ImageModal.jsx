import React, { Component } from "react";
import Dropzone from "react-dropzone";
import ReactAvatarEditor from "react-avatar-editor";

import { Button } from "reactstrap";
import { observer } from "mobx-react";
import { observable } from "mobx";
import { translate } from "common/methods/translations";

import Modal from "elements/Modal/Modal";
import styles from "./ImageModal.scss";

@observer
class ImageModal extends Component {
  @observable file = null;
  @observable errorNotice = null;

  constructor() {
    super();
    this.renderUploadedFiles = this.renderUploadedFiles.bind(this);
  }

  saveButtonClick() {
    const { handleYesButtonClick } = this.props;
    handleYesButtonClick(this.file ? this.file[0] : null);
    this.file = null;
    this.errorNotice = null;
  }

  cancelButtonClick() {
    const { handleNoButtonClick } = this.props;
    handleNoButtonClick();
    this.file = null;
    this.errorNotice = null;
  }

  saveCancelButtons() {
    return (
      <div>
        <Button color="primary" onClick={this.saveButtonClick.bind(this)}>
          {translate("globals.create")}
        </Button>
        <Button color="secondary" onClick={this.cancelButtonClick.bind(this)}>
          {translate("globals.cancel")}
        </Button>
      </div>
    );
  }

  onDropRejected = e => {
    console.log(e);
    this.errorNotice = translate("globals.maxUpload", { size: "2 MB" });
  };

  onDrop(file) {
    this.file = file;
    this.errorNotice = null;
  }

  renderUploadedFiles() {
    if (this.file && this.file[0]) {
      return (
        <ReactAvatarEditor
          ref={this.props.setEditorRef}
          border={1}
          width={100}
          height={100}
          image={this.file[0]}
          borderRadius={50}
        />
      );
    }
    return this.errorNotice ? (
      <p className={styles.SizeError}>{this.errorNotice}</p>
    ) : (
      <p>
        {translate("myProfile.personalInformation.uploadTitle")}.{" "}
        {translate("myProfile.personalInformation.uploadSubtitle")}
      </p>
    );
  }

  render() {
    return (
      <Modal
        ModalFooterContent={this.saveCancelButtons.bind(this)}
        {...this.props}
        className={"NewFileModal"}
      >
        <Dropzone
          className="dropzone"
          accept="image/jpeg, image/png"
          onDrop={this.onDrop.bind(this)}
          onDropRejected={this.onDropRejected}
          maxSize={2097152}
        >
          {this.renderUploadedFiles()}
        </Dropzone>
      </Modal>
    );
  }
}

export default ImageModal;
