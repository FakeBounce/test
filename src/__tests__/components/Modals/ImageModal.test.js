import React from "react";
import { shallow, configure } from "enzyme/build";
import Adapter from "enzyme-adapter-react-16/build";
import ImageModal from "components/Modals/ImageModal/ImageModal";

configure({ adapter: new Adapter() });

const handleYes = jest.fn();
const handleNo = jest.fn();
const setEditor = jest.fn();

const defaultTestingProps = {
  isOpen: false,
  title: "Title",
  handleYesButtonClick: handleYes,
  handleNoButtonClick: handleNo,
  setEditorRef: setEditor,
  toggle: () => jest.fn()
};

describe("Testing ImageModal component", () => {
  it("renders as expected without parameters", () => {
    const wrapper = shallow(<ImageModal />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders as expected with parameters (as closed)", () => {
    const wrapper = shallow(<ImageModal {...defaultTestingProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders as expected with parameters (as open)", () => {
    const wrapper = shallow(<ImageModal {...defaultTestingProps} isOpen />);
    expect(wrapper).toMatchSnapshot();
  });

  it("behave correctly when calling props", () => {
    const wrapper = shallow(<ImageModal {...defaultTestingProps} />);

    wrapper.instance().saveButtonClick();
    expect(handleYes.mock.calls.length).toBe(1);
      // expect(handleYes.mock.calls.length).toBeCalledWith(false);
  });

  it("behave correctly when props are invalid", () => {
      // @TODO how can we check setEditorRef as it doesn't throw an error ?
    const wrapper = shallow(
      <ImageModal
        {...defaultTestingProps}
        handleYesButtonClick={true}
        handleNoButtonClick={true}
        setEditorRef={true}
      />
    );
    try {
      wrapper.instance().saveButtonClick();
      // Fail test if above expression doesn't throw anything.
      expect(true).toBe(false);
    } catch (e) {
      expect(e).toBeInstanceOf(TypeError);
      expect(e.message).toBe("handleYesButtonClick is not a function");
    }
    try {
      wrapper.instance().cancelButtonClick();
      // Fail test if above expression doesn't throw anything.
      expect(true).toBe(false);
    } catch (e) {
      expect(e).toBeInstanceOf(TypeError);
      expect(e.message).toBe("handleNoButtonClick is not a function");
    }
  });
});
