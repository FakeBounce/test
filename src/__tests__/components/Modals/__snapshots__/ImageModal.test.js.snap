// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ImageModal component renders as expected with parameters (as closed) 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ImageModal
    handleNoButtonClick={[Function]}
    handleYesButtonClick={[Function]}
    isOpen={false}
    setEditorRef={[Function]}
    title="Title"
    toggle={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "ModalFooterContent": [Function],
      "children": <n
        accept="image/jpeg, image/png"
        className="dropzone"
        disableClick={false}
        disablePreview={false}
        disabled={false}
        getDataTransferItems={[Function]}
        inputProps={Object {}}
        maxSize={2097152}
        minSize={0}
        multiple={true}
        onDrop={[Function]}
        onDropRejected={[Function]}
        preventDropOnDocument={true}
>
        <p>
                
                .
                 
                
        </p>
</n>,
      "className": "NewFileModal",
      "handleNoButtonClick": [Function],
      "handleYesButtonClick": [Function],
      "isOpen": false,
      "setEditorRef": [Function],
      "title": "Title",
      "toggle": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "accept": "image/jpeg, image/png",
        "children": <p>
          
          .
           
          
</p>,
        "className": "dropzone",
        "disableClick": false,
        "disablePreview": false,
        "disabled": false,
        "getDataTransferItems": [Function],
        "inputProps": Object {},
        "maxSize": 2097152,
        "minSize": 0,
        "multiple": true,
        "onDrop": [Function],
        "onDropRejected": [Function],
        "preventDropOnDocument": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "",
            ".",
            " ",
            "",
          ],
        },
        "ref": null,
        "rendered": Array [
          "",
          ".",
          " ",
          "",
        ],
        "type": "p",
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "ModalFooterContent": [Function],
        "children": <n
          accept="image/jpeg, image/png"
          className="dropzone"
          disableClick={false}
          disablePreview={false}
          disabled={false}
          getDataTransferItems={[Function]}
          inputProps={Object {}}
          maxSize={2097152}
          minSize={0}
          multiple={true}
          onDrop={[Function]}
          onDropRejected={[Function]}
          preventDropOnDocument={true}
>
          <p>
                    
                    .
                     
                    
          </p>
</n>,
        "className": "NewFileModal",
        "handleNoButtonClick": [Function],
        "handleYesButtonClick": [Function],
        "isOpen": false,
        "setEditorRef": [Function],
        "title": "Title",
        "toggle": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accept": "image/jpeg, image/png",
          "children": <p>
            
            .
             
            
</p>,
          "className": "dropzone",
          "disableClick": false,
          "disablePreview": false,
          "disabled": false,
          "getDataTransferItems": [Function],
          "inputProps": Object {},
          "maxSize": 2097152,
          "minSize": 0,
          "multiple": true,
          "onDrop": [Function],
          "onDropRejected": [Function],
          "preventDropOnDocument": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "",
              ".",
              " ",
              "",
            ],
          },
          "ref": null,
          "rendered": Array [
            "",
            ".",
            " ",
            "",
          ],
          "type": "p",
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Array [],
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Testing ImageModal component renders as expected with parameters (as open) 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ImageModal
    handleNoButtonClick={[Function]}
    handleYesButtonClick={[Function]}
    isOpen={true}
    setEditorRef={[Function]}
    title="Title"
    toggle={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "ModalFooterContent": [Function],
      "children": <n
        accept="image/jpeg, image/png"
        className="dropzone"
        disableClick={false}
        disablePreview={false}
        disabled={false}
        getDataTransferItems={[Function]}
        inputProps={Object {}}
        maxSize={2097152}
        minSize={0}
        multiple={true}
        onDrop={[Function]}
        onDropRejected={[Function]}
        preventDropOnDocument={true}
>
        <p>
                
                .
                 
                
        </p>
</n>,
      "className": "NewFileModal",
      "handleNoButtonClick": [Function],
      "handleYesButtonClick": [Function],
      "isOpen": true,
      "setEditorRef": [Function],
      "title": "Title",
      "toggle": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "accept": "image/jpeg, image/png",
        "children": <p>
          
          .
           
          
</p>,
        "className": "dropzone",
        "disableClick": false,
        "disablePreview": false,
        "disabled": false,
        "getDataTransferItems": [Function],
        "inputProps": Object {},
        "maxSize": 2097152,
        "minSize": 0,
        "multiple": true,
        "onDrop": [Function],
        "onDropRejected": [Function],
        "preventDropOnDocument": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "",
            ".",
            " ",
            "",
          ],
        },
        "ref": null,
        "rendered": Array [
          "",
          ".",
          " ",
          "",
        ],
        "type": "p",
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "ModalFooterContent": [Function],
        "children": <n
          accept="image/jpeg, image/png"
          className="dropzone"
          disableClick={false}
          disablePreview={false}
          disabled={false}
          getDataTransferItems={[Function]}
          inputProps={Object {}}
          maxSize={2097152}
          minSize={0}
          multiple={true}
          onDrop={[Function]}
          onDropRejected={[Function]}
          preventDropOnDocument={true}
>
          <p>
                    
                    .
                     
                    
          </p>
</n>,
        "className": "NewFileModal",
        "handleNoButtonClick": [Function],
        "handleYesButtonClick": [Function],
        "isOpen": true,
        "setEditorRef": [Function],
        "title": "Title",
        "toggle": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accept": "image/jpeg, image/png",
          "children": <p>
            
            .
             
            
</p>,
          "className": "dropzone",
          "disableClick": false,
          "disablePreview": false,
          "disabled": false,
          "getDataTransferItems": [Function],
          "inputProps": Object {},
          "maxSize": 2097152,
          "minSize": 0,
          "multiple": true,
          "onDrop": [Function],
          "onDropRejected": [Function],
          "preventDropOnDocument": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "",
              ".",
              " ",
              "",
            ],
          },
          "ref": null,
          "rendered": Array [
            "",
            ".",
            " ",
            "",
          ],
          "type": "p",
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Array [],
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Testing ImageModal component renders as expected without parameters 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ImageModal />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "ModalFooterContent": [Function],
      "children": <n
        accept="image/jpeg, image/png"
        className="dropzone"
        disableClick={false}
        disablePreview={false}
        disabled={false}
        getDataTransferItems={[Function]}
        inputProps={Object {}}
        maxSize={2097152}
        minSize={0}
        multiple={true}
        onDrop={[Function]}
        onDropRejected={[Function]}
        preventDropOnDocument={true}
>
        <p>
                
                .
                 
                
        </p>
</n>,
      "className": "NewFileModal",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "accept": "image/jpeg, image/png",
        "children": <p>
          
          .
           
          
</p>,
        "className": "dropzone",
        "disableClick": false,
        "disablePreview": false,
        "disabled": false,
        "getDataTransferItems": [Function],
        "inputProps": Object {},
        "maxSize": 2097152,
        "minSize": 0,
        "multiple": true,
        "onDrop": [Function],
        "onDropRejected": [Function],
        "preventDropOnDocument": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "",
            ".",
            " ",
            "",
          ],
        },
        "ref": null,
        "rendered": Array [
          "",
          ".",
          " ",
          "",
        ],
        "type": "p",
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "ModalFooterContent": [Function],
        "children": <n
          accept="image/jpeg, image/png"
          className="dropzone"
          disableClick={false}
          disablePreview={false}
          disabled={false}
          getDataTransferItems={[Function]}
          inputProps={Object {}}
          maxSize={2097152}
          minSize={0}
          multiple={true}
          onDrop={[Function]}
          onDropRejected={[Function]}
          preventDropOnDocument={true}
>
          <p>
                    
                    .
                     
                    
          </p>
</n>,
        "className": "NewFileModal",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accept": "image/jpeg, image/png",
          "children": <p>
            
            .
             
            
</p>,
          "className": "dropzone",
          "disableClick": false,
          "disablePreview": false,
          "disabled": false,
          "getDataTransferItems": [Function],
          "inputProps": Object {},
          "maxSize": 2097152,
          "minSize": 0,
          "multiple": true,
          "onDrop": [Function],
          "onDropRejected": [Function],
          "preventDropOnDocument": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "",
              ".",
              " ",
              "",
            ],
          },
          "ref": null,
          "rendered": Array [
            "",
            ".",
            " ",
            "",
          ],
          "type": "p",
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Array [],
  Symbol(enzyme.__childContext__): null,
}
`;
